base/shared/src/main/scala/scalaz/data/disjunction.scala

```diff
@@ -69,11 +69,13 @@ trait DisjunctionInstances {
     })
 
   implicit def disjunctionEq[A, B](implicit A: Eq[A], B: Eq[B]): Eq[A \/ B] =
-    instanceOf[EqClass[A \/ B]] {
-      case (-\/(a1), -\/(a2)) => A.equal(a1, a2)
-      case (\/-(b1), \/-(b2)) => B.equal(b1, b2)
-      case _                  => false
-    }
+    instanceOf(new EqAnyRef[A \/ B]() {
```

Comment:
you can keep the anonymous lambda syntax here (same of other methods). See https://github.com/scalaz/scalaz/compare/series/8.0.x...jbgi:eq-ref-template?expand=1#diff-104602adaa1126e4668c0c18f3605f8cR40